consensus/src/replica.rs

```diff
@@ -373,6 +375,15 @@ impl<A: ConsensusApi<TYPES>, TYPES: NodeTypes> Replica<A, TYPES> {
                 .metrics
                 .invalid_qc_views
                 .add_point(consensus.invalid_qc as f64);
+            #[allow(clippy::cast_possible_truncation)]
+            let included_txn_size = included_txns_set.iter().fold(0, |total, txn| {
+                total + bincode_opts().serialized_size(&txn).unwrap_or(0) as usize
+            });
+            consensus.pending_transaction_mem -= included_txn_size;
```

Comment:
Should we add something like `update(delta: i64)` to the `Gauge` trait so we don't have to store the current value of the gauge separately for metrics like this?

Reply:
Yeah i've been thinking of this, I'm going to add it.  I thought it had to do with the Prometheus gauge type not having any kind of update but I checked and it expose, get, add and subtract.